bgpd/bgp_route.h

```diff
@@ -806,7 +806,8 @@ extern void bgp_withdraw(struct peer *peer, const struct prefix *p,
 
 /* for bgp_nexthop and bgp_damp */
 extern void bgp_process(struct bgp *bgp, struct bgp_dest *dest,
-			struct bgp_path_info *pi, afi_t afi, safi_t safi);
```

Comment:
instead of changing this function, maybe add a "process_early" version, that you can use in the specific places where you want the "early" behavior?

Reply:
adding "early," though, allows you to control the behavior on a route-by-route basis, perhaps even (eventually) using a route-map or some policy to let some routes be processed before others (?) ...

Reply:
I agree that there needs to be an API, and that it needs to be called - I'm just thinking it'd be better to have a dedicated api, that would only involve changing the places where it's needed. This change requires many callers to be changed, so that they can say "false"...

Reply:
I added a bgp_process_early  I believe it solves Marks issues

Reply:
that's great, thanks.
so ... could we have a comment, even a one-liner, to say that the "early" version puts the dest at the front of the list?